File: Java/ejemplo03.java
import java.util.ArrayList; 
public class ejemplo3 {

	public static void main(String[] args) {
		ArrayList<Integer> x; 
		
		ArrayList<Integer> y;
		
		ArrayList<Integer> z;
		
		int[] vals= new int[6];
		
		for(int i=0;i<6;i++) {
			vals[i]=i;
		}
		
		int[] inter1= {9, 29, 54, 79, 94, 99};
		
		int n=99;
		
		x=setVar(vals,inter1,n);
		
		for(int i=0;i<4;i++) {
			vals[i]=i;
		}
		
		int[] inter2= {24, 54, 79, 99};
		
		y=setVar(vals,inter2,n);
		
		for(int i=0;i<3;i++) {
			vals[i]=i;
		}
		
		int[] inter3= {2,7,9};
		
		n=9;
		
		z=setVar(vals,inter3,n);
		
		funcEfectividad(x,y,z);
		

	}

	
	public static ArrayList<Integer> setVar(int [] vals,int [] inter,int n){
		ArrayList<Integer> list = new ArrayList<Integer>(); 
		int interIndex=0;
		for(int i=0;i<=n;i++) {
			if(i<=inter[interIndex]) {
				
			}
			else {
				interIndex+=1;
			}
			list.add(vals[interIndex]);
		}
		
		return list;
	}
	
	public static void funcEfectividad(ArrayList<Integer> x,ArrayList<Integer> y,ArrayList<Integer> z) {
		int [] randX = {43, 96, 57, 53, 14, 3, 33, 40};
		int [] randY = {22, 50, 13, 36, 91, 58, 45, 43};
		int [] randZ = {1, 8, 0, 2, 7, 6, 1, 3};
		int xi,yi,zi;
		double wi=0;
		System.out.println("Rand X || Val X || Rand Y || Val Y || Rand Z || Val Z || X+4Y+3Z");
		
		for(int i=0;i<8;i++) {
			xi=getVar(randX[i],x);
			yi=getVar(randY[i],y);
			zi=getVar(randZ[i],z);
			
			wi=xi+4*yi+3*zi;
			
			System.out.println(randX[i]+" || "+xi+" || "+randY[i]+" || "+yi+" || "+randZ[i]+" || "+zi+" || "+wi);
		}

	}
	
	public static int getVar(int num, ArrayList<Integer> y) {
		return y.get(num);
	}
}

File: CodigoPY/ejemplo03.py
import matplotlib.pyplot as plt
from numpy import random
from typing import List

class Ejemplo03:
    """
    clase para el desarollo del ejemplo #03
    """
    def setVar(self, vals: List[int | float], inters: List[int | float], n: int) -> List[int | float]:
        """
        Metodo para definir los valores de y
        DESCRIPCION DE LA FUNCION SETVAR
        Definimos los valores de y dados por la distribucion mostrada en la funcion
        Para esto creamos dos listas:
            1. vals: valores que tomara y en cada intervalo
            2. inters: Valor superior del intervalo de numeros que toman los valores
        Importante que vals y clases tengan la misma cantidad de elementos.
        Ejemplo:
            valores     |     intervalo
               0        |        0-15
               1        |       16-40
               2        |       40-83
               3        |       83-99
        :param vals: tomaria los valores 0 a 3 'y'
        :type vals: List[int | float]
        :param inters: tomaria los valores 15, 40, 83 y 99
        :type inters: List[int | float]
        :param n: Numero de valores de la lista
        :type n: int
        :return lista con los valores
        :rtype List[int|float]
        """
        lista: List[int | float] = []
        interIndex: int = 0
        for i in range(n+1):
            if i<= inters[interIndex]:
                pass
            else:
                interIndex +=1
            lista.append(vals[interIndex])
        return lista

    def getVar(self, num: int, y: List[int | float]) -> int | float:
        """
        Obtener el valor e Y
        :param num: indice a buscar
        :type num: int
        :param y: lista con los valores
        :type y: List[int | float]
        :return: el valor deseado
        :rtype int | float
        """
        return y[num]


    def present(self, w: List[int | float]) -> None:
        """
        Imprime los valores en pantalla en un grafico
        :param w: lista con los valores a graficar
        :return: nada
        :rtype None
        """
        plt.hist(w, bins=[5,10,15,20,25,32])
        plt.title("Histograma de Distribución de W")
        plt.show()


    def randoms(self, n: int,maxi:int) -> List[int]:
        """
        metodo para generar los valores pseudo aleatorios enteros
        :param n: tamano de la lista a generar
        :type n: int
        :param maxi: valor maximo de los valores pseudo aleatorios
        :type maxi: int
        :return: lista de numeros pseudo aleatorios
        :rtype List[int]
        """
        lista: List[int] = []
        for i in range(n):
            lista.append(random.randint(0,maxi))
        return lista

    def funcEfectividad(self, x: List[int] ,y: List[int] ,z: List[int] ,n: int) -> None:
        """
        metodo para medir la efectividad
        :param x: Lista con los valores de X
        :type x:
        :param y: Lista con los valores de Y
        :type y: List[int]
        :param z: Lista con los valores de Z
        :type z: List[int]
        :param n: cantidad de veces a ejecutar este analisis
        :type n: int
        :return: Nada
        :rtype None
        """
        #Utilizaremos los nuemeros aleatorios del ejemplo:
        randX: List[int]  = self.randoms(n, maxi=99)
        randY: List[int]  = self.randoms(n,maxi=99)
        randZ: List[int]  = self.randoms(n,maxi=9)

        #Inicializamos la data:
        w: List[int] = []
        for i in range(n):
            xi: int = self.getVar(randX[i], x)
            yi: int = self.getVar(randY[i], y)
            zi: int = self.getVar(randZ[i], z)

            wi: int = xi+4*yi+3*zi

            w.append(wi)

        self.present(w)


    def run(self) -> None:
        """
        Metodo para ejecutar el ejemplo #3
        Definicion de variables para la ecuacion W = X + 4Y + 3Z
        X definida en base al enunciado
        :return: nada
        :rtype None
        """

        x = self.setVar(
            vals=[x+1 for x in range(6)],
            inters=[9, 29, 54, 79, 94, 99],
            n = 99
        )

        y = self.setVar(
            vals=[x+1 for x in range(4)],
            inters=[24, 54, 79, 99],
            n = 99
        )

        z = self.setVar(
            vals=[x+1 for x in range(3)],
            inters=[2,7,9],
            n = 9
        )

        n= int(input('¿Cuantas veces desea correr la simulacion de w?: '))
        self.funcEfectividad(x,y,z,n)
